clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -730,6 +730,15 @@ void CodeGenFunction::EmitOpenCLKernelMetadata(const FunctionDecl *FD,
         llvm::ConstantAsMetadata::get(Builder.getInt32(1))};
     Fn->setMetadata("stall_enable", llvm::MDNode::get(Context, AttrMDArgs));
   }
+
+  if (const SYCLIntelFPGAMaxConcurrencyAttr *A =
+          FD->getAttr<SYCLIntelFPGAMaxConcurrencyAttr>()) {
+    const auto *CE = dyn_cast<ConstantExpr>(A->getNThreadsExpr());
```

Comment:
you are missing an assert here before ArgVal.
assert(CE && "Not an integer constant expression");

Reply:
```suggestion
    const auto *CE = cast<ConstantExpr>(A->getNThreadsExpr());
```
No need for a `dyn_cast<>` followed by an `assert`.